lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -2369,7 +2397,16 @@ public function createEntity($className, array $data, &$hints = [])
             $this->originalEntityData[$oid][$field] = $newValue;
             $association->setValue($entity, $newValue);
 
-            if ($association->getInversedBy() && $association instanceof OneToOneAssociationMetadata) {
+            if (
```

Comment:
I think this entire block seems wrong and should be removed. The code in this block overwrites the association field of the related entity, which is wrong.

Reply:
This code cannot be fixed in here